install_scripts/app.py

```diff
@@ -67,11 +67,39 @@ def get_replicated_version(replicated_channel=None,
     return replicated_version
 
 
-@app.route('/')
-@app.route('/<replicated_channel>')
-def get_replicated_one_point_two(replicated_channel=None):
-    replicated_channel = replicated_channel if replicated_channel else 'stable'
-    kwargs = helpers.template_args(channel_name=replicated_channel)
+@app.route('/', defaults={'path': ''})
```

Comment:
I like the 404 handler, but it would be better to avoid duplicating the 1.2 script code in four places. What about something like this?
```
@app.route('/', defaults={'replicated_channel': 'stable'})
@app.route('/stable', defaults={'replicated_channel': 'stable'})
@app.route('/unstable', defaults={'replicated_channel': 'unstable'})
@app.route('/beta', defaults={'replicated_channel': 'beta'})
def get_replicated_one_point_two(replicated_channel):
        kwargs = helpers.template_args(channel_name=replicated_channel)
        kwargs['pinned_docker_version'] = '1.12.3'
        response = render_template('replicated-1.2.sh', **kwargs)
        return Response(response, mimetype='text/x-shellscript')

@app.route('/<path:path>')
def catch_all(path):
        kwargs = helpers.template_args(path=path)
        response = render_template('resolve-route.sh', **kwargs)
        return Response(response, mimetype='text/x-shellscript')
```

Reply:
That works (tested to confirm)! I couldn't quite get route linking working like which is why I broke out the replicated channel routes. I'll make that change to the PR. Thanks!